kani-compiler/src/kani_middle/transform/loop_contracts.rs

```diff
@@ -156,6 +192,326 @@ impl LoopContractPass {
         body.replace_statements(&SourceInstruction::Terminator { bb: bb_idx }, stmts);
     }
 
+    fn get_user_defined_variables(&self, body: &MutableBody) -> Vec<usize> {
```

Comment:
Could you please comments to newly introduced functions. Here, I am specifically looking to understand what is meant by "user" defined variables.